BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BTConnectToThread.java

```diff
@@ -40,21 +54,46 @@ public BTConnectToThread(BtConnectToCallback Callback, BluetoothDevice device, U
     }
     public void run() {
         printe_line("Starting to connect");
-        if(mSocket != null && callback != null) {
+        if (mSocket == null || callback == null) {
+            return;
+        }
+        try {
+            mSocket.connect();
+            //return success
+            if (mBTHandShakeSocketTread == null) {
+                HandShakeTimeOutTimer.start();
+
+                mBTHandShakeSocketTread = new BTHandShakeSocketTread(mSocket, mHandler);
+                mBTHandShakeSocketTread.start();
+                mBTHandShakeSocketTread.write(mInstanceString.getBytes());
+            }
+        } catch (IOException e) {
+            printe_line("socket connect failed: " + e.toString());
             try {
-                mSocket.connect();
-                //return success
-                callback.Connected(mSocket,mPeerId,mPeerName,mPeerAddress);
-            } catch (IOException e) {
-                printe_line("socket connect failed: " + e.toString());
-                try {
-                    mSocket.close();
-                } catch (IOException ee) {
-                    printe_line("closing socket 2 failed: " + ee.toString());
-                }
-                callback.ConnectionFailed(e.toString(),mPeerId,mPeerName,mPeerAddress);
+                mSocket.close();
+            } catch (IOException ee) {
+                printe_line("closing socket 2 failed: " + ee.toString());
             }
+            callback.ConnectionFailed(e.toString(), mPeerId, mPeerName, mPeerAddress);
+        }
+
+    }
+
+    private void HandShakeOk() {
+        HandShakeTimeOutTimer.cancel();
+        mBTHandShakeSocketTread = null;
+
+        callback.Connected(mSocket,mPeerId,mPeerName,mPeerAddress);
+    }
+
+    private void HandShakeFailed(String reason) {
+        HandShakeTimeOutTimer.cancel();
+        BTHandShakeSocketTread tmp = mBTHandShakeSocketTread;
+        mBTHandShakeSocketTread = null;
+        if(tmp != null) {
+            tmp.CloseSocket();
         }
+        callback.ConnectionFailed("handshake: " + reason, mPeerId, mPeerName, mPeerAddress);
     }
 
     private void printe_line(String message){
```

Comment:
/printe_line/print_line

But the more important point than the spelling error is that we should just use Android logging as-is. There is no need for this redirection.